atom/browser/api/atom_api_session.cc

```diff
@@ -237,12 +239,24 @@ void SetProxyInIO(net::URLRequestContextGetter* getter,
   RunCallbackInUI(callback);
 }
 
+void PassVerificationResult(
+    scoped_refptr<AtomCertVerifier::CertVerifyRequest> request,
+    bool success) {
+  int result = net::OK;
+  if (!success)
+    result = net::ERR_FAILED;
+  request->ContinueWithResult(result);
+}
+
 }  // namespace
 
 Session::Session(AtomBrowserContext* browser_context)
     : browser_context_(browser_context) {
   AttachAsUserData(browser_context);
 
+  // Observe Browser to get certificate verification notification.
+  Browser::Get()->AddObserver(this);
```

Comment:
`Browser` is application-wide, not session-wide. You can probably add a delegate class to `AtomCertVerifier`.